### DISCUSSION TITLE
Automatic1111 new feature

### DISCUSSION THREAD

**Discussion by PaulQuiDev** [open-webui/open-webui#4471] [Ideas]
Hello, it's a great opportunity to have made possible the connection between open-webui and automatic1111! 
However, some very important parameters for image generation are missing, such as the CFG Scale or the sampler. Would it be possible to add them in the image generation parameters? 

Also, the automatic1111 api has changed a bit in v1.10.1 and the Sampler Scheduler autocorrect and ask. It sends back an alert, but it still works.
<code>Model loaded in 19.7s (load weights from disk: 1.4s, create model: 2.1s, apply weights to model: 15.2s, apply float(): 0.3s, calculate empty prompt: 0.7s).
<b>WARNING:root:Sampler Scheduler autocorrection: "DPM++ 2M" -> "DPM++ 2M", "None" -> "Automatic"</b>
100%|███████████████████████████████████████████| 20/20 [01:06<00:00,  3.33s/it]</code>

Now to change the sampler I have to change the default sampler in the stable-diffusion-webui folder.
in stable-diffusion-webui/modules/api/models.py 
l104 <code>{"key": "sampler_index", "type": str, "default": "NameOfSampler"},</code>
change NameOfSampler to the desired sampler

**Comment by Rotoslider** [comment]
{"key": "sampler_index", "type": str, "default": "DPM++ 3M SDE"}
produces:
WARNING:root:Sampler Scheduler autocorrection: "DPM++ 3M SDE" -> "DPM++ 3M SDE", "None" -> "Automatic"00:00,
How can we set default in the command line instead of modifying models.py

It is applying the DPM++ #M SDE but it sets the Schedule Type to Exponential I need it to be Karras under the sd_samplers_kdiffusion.py. 
('DPM++ 3M SDE', 'sample_dpmpp_3m_sde', ['k_dpmpp_3m_sde'], {'scheduler': 'exponential', 'discard_next_to_last_sigma': True, "brownian_noise": True}),
There needs to be a better way to set this up for cmd line use.

**Reply by PaulQuiDev** [reply]
I haven't found any other way that's less painful, but you can change the scheduler by adding the <code>{"key": "scheduler", "type": str, "default": "Name_of_scheduler"},</code> command in stable-diffusion-webui/modules/api/models.py. 
after line 111 like this 
<code>{"key": "force_task_id", "type": str, "default": None},
        {"key": "infotext", "type": str, "default": None},
        <b>{"key": "scheduler", "type": str, "default": "<I>Uniform</I>"},</b></code>

I would have liked to be able to choose these options from the openweb interface. =')

**Reply by an80sPWNstar** [reply]
Has there been any updates to this? I can generate one image just fine but trying to use Loras is dang near impossible and breaks the API connection to the a1111 server in that chat. If I want to create another image, I have to open a new chat.
